cdk/kittyhawk/src/certificate.ts

```diff
@@ -10,26 +10,24 @@ export class Certificate extends Construct {
     if (props.ingress) {
       // We want to generate a certificate for each host
       for (const h of props.ingress) {
-        // Regex to compute the apex domain
-        const apex_domain = h.host.match(/[\w-]+\.[\w]+$/g);
-        if (apex_domain != null) {
-          let host_string = apex_domain[0].split('.').join('-');
-          new CertApiObject(this, `certificate-${appname}-${host_string}`, {
-            metadata: {
-              name: host_string,
+        let hostString: string = domainToCertName(h.host, h.isSubdomain);
+        new CertApiObject(this, `certificate-${appname}-${hostString}`, {
+          metadata: {
+            name: hostString,
+          },
+          spec: {
+            secretName: hostString.concat('-tls'),
+            dnsNames: [`${removeSubdomain(h.host, h.isSubdomain)}`, `*.${removeSubdomain(h.host, h.isSubdomain)}`],
+            issuerRef: {
+              name: 'wildcard-letsencrypt-prod',
+              kind: 'ClusterIssuer',
+              group: 'cert-manager.io',
             },
-            spec: {
-              secretName: host_string.concat('-tls'),
-              dnsNames: [`${apex_domain[0]}`, `*.${apex_domain[0]}`],
-              issuerRef: {
-                name: 'wildcard-letsencrypt-prod',
-                kind: 'ClusterIssuer',
-                group: 'cert-manager.io',
-              },
-            },
-          });
-        } else {throw `Certificate construction failed: apex domain regex failed on ${h}`;}
+          },
+        });
       }
+    } else {
+      throw new Error('Cannot generate certificate if props.ingress is undefined.');
```

Comment:
See optional ingress prop comment, if that becomes required then this check becomes static

Reply:
Resolved from ingress prop discussion.